### PR TITLE
KAFKA-17323: Document UINT16 and COMPACT_RECORDS in Protocol Guide

### PR DESCRIPTION
**Pull request by ivanyu**
As these types are used in the protocol.

*More detailed description of your change,
if necessary. The PR title and PR message become
the squashed commit message, so use a separate
comment to ping reviewers.*

*Summary of testing strategy (including rationale)
for the feature or bug fix. Unit and/or integration
tests are expected for any behaviour change and
system tests should be considered for larger changes.*

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: clients/src/main/java/org/apache/kafka/common/protocol/types/Type.java [hunk 1]
```diff
@@ -1113,7 +1113,7 @@ public String documentation() {
     private static String toHtml() {
         DocumentedType[] types = {
             BOOLEAN, INT8, INT16, INT32, INT64,
-            UNSIGNED_INT32, VARINT, VARLONG, UUID, FLOAT64,
```
Comment (chia7712) [inline clients/src/main/java/org/apache/kafka/common/protocol/types/Type.java:1116]: Maybe we should document "all" types in the protocol guide even though some types are unused yet.

**Review by chia7712 — COMMENTED** [review]
@ivanyu thanks for this patch

File: clients/src/main/java/org/apache/kafka/common/protocol/types/Type.java (hunk 1 above)
Comment (ivanyu) [reply clients/src/main/java/org/apache/kafka/common/protocol/types/Type.java:1116]: This makes sense. I added `UINT16` and `COMPACT_RECORDS` without removing anything.
I skipped the tagged fields though, as referencing to them also needs fixing throughout the page and it's probably better be done separately.

**Review by chia7712 — APPROVED** [review]
LGTM

**Comment by ivanyu** [comment]
Thank you @chia7712. If you have a moment, could you please also look at https://github.com/apache/kafka/pull/16867 and https://github.com/apache/kafka/pull/17122? They also touch the protocol documentation and are short.

**Comment by chia7712** [comment]
> could you please also look at https://github.com/apache/kafka/pull/16867 and https://github.com/apache/kafka/pull/17122? They also touch the protocol documentation and are short.

sure :smile:

**Pull request merged**